drivers/ethernet/eth_stm32_hal.c

```diff
@@ -847,8 +847,8 @@ static void rx_thread(void *arg1, void *unused1, void *unused2)
 			}
 		} else if (res == -EAGAIN) {
 			/* semaphore timeout period expired, check link status */
-			hal_ret = read_eth_phy_register(&dev_data->heth,
-				    PHY_ADDR, PHY_BSR, (uint32_t *) &status);
+			hal_ret = read_eth_phy_register(config->phy_dev, &dev_data->heth, PHY_ADDR,
```

Comment:
Sure! I will update that along with some twister CI error fixes, it really feels wrong to pass two parts of one dev individually when it could be the device itself and the functions still does not lose any coherence!